src/number_theory/pythagorean_triples.lean

```diff
@@ -534,6 +534,54 @@ begin
     { split, { ring }, rw int.gcd_comm, exact coprime_pow_two_sub_mul co pp } }
 end
 
+/-- by assuming `x` is odd and `z` is positive we get a slightly more precise classification of
+the pythagorean triple `x ^ 2 + y ^ 2 = z ^ 2`-/
+theorem coprime_classification' {x y z : ℤ} (h : pythagorean_triple x y z)
+  (h_coprime : int.gcd x y = 1) (h_parity : x % 2 = 1) (h_pos : 0 < z) :
+  ∃ m n,  x = m ^ 2 - n ^ 2
+        ∧ y = 2 * m * n
+        ∧ z = m ^ 2 + n ^ 2
+        ∧ int.gcd m n = 1
+        ∧ ((m % 2 = 0 ∧ n % 2 = 1) ∨ (m % 2 = 1 ∧ n % 2 = 0))
+        ∧ 0 ≤ m :=
+begin
+  obtain ⟨m, n, ht1, ht2, ht3, ht4⟩ :=
+    pythagorean_triple.coprime_classification.mp (and.intro h h_coprime),
+  by_cases hm_sign : 0 ≤ m,
```

Comment:
I would have been tempted to write `cases le_or_lt 0 m with hm hm,` giving me `m < 0` rather than the more annoying to use `not 0 <= m`. but you never use it anyway :-)

Reply:
@paulvanwamelen what do you think of this suggestion?

Reply:
I thought this was just a comment, not a suggestion. I'll change.